simphony_mayavi/sources/slim_cuds_source.py

```diff
@@ -0,0 +1,232 @@
+import itertools
+import numpy
+from traits.api import TraitError
+
+from simphony.core.cuba import CUBA
+from simphony.cuds import ABCMesh, ABCParticles, ABCLattice
+from simphony.io.h5_mesh import H5Mesh
+from simphony.testing.utils import dummy_cuba_value
+from simphony_mayavi.cuds.vtk_lattice import VTKLattice
+from simphony_mayavi.cuds.vtk_mesh import VTKMesh
+from simphony_mayavi.cuds.vtk_particles import VTKParticles
+from simphony_mayavi.sources.cuds_source import CUDSSource
+
+
+class SlimCUDSSource(CUDSSource):
+    """Overrides CUDSSource to provide a source optimized for lower memory
+    consumption, sacrificing access speed.
+    This is achieved by retrieving data as they are requested by the user,
+    instead of preemptively load all available data.
+    """
+
+    def _point_scalars_name_changed(self, value):
+        super(CUDSSource, self)._point_scalars_name_changed(value)
+        self.update()
+
+    def _point_vectors_name_changed(self, value):
+        super(CUDSSource, self)._point_vectors_name_changed(value)
+        self.update()
+
+    def _cell_scalars_name_changed(self, value):
+        super(CUDSSource, self)._cell_scalars_name_changed(value)
+        self.update()
+
+    def _cell_vectors_name_changed(self, value):
+        super(CUDSSource, self)._cell_vectors_name_changed(value)
+        self.update()
+
+    def _set_cuds(self, cuds):
+
+        # Before refreshing the VTK CUDS object, we need to set the content
+        # of the available data in the cuds. Normally this is done by the
+        # VTKDataSource _from_ the VTK data, but our VTK data will not contain
+        # all the keys. We are therefore required to update them here.
+        all_lists = (
+            self._point_scalars_list,
+            self._point_vectors_list,
+            self._cell_scalars_list,
+            self._cell_vectors_list)
+
+        for lst, keys in zip(all_lists, _available_keys(cuds)):
+            lst[:] = sorted([key.name for key in keys])
```

Comment:
Need to add an empty string here for the user to unselect the data attribute:
https://github.com/enthought/mayavi/blob/master/mayavi/sources/vtk_data_source.py#L345